component/src/main/java/org/wso2/extension/siddhi/io/jms/sink/JMSPublisher.java

```diff
@@ -19,74 +19,76 @@
-    private static final String BYTE_ARRAY_MESSAGE_TYPE = "ByteMessage";
-
     private Map<String, String> jmsProperties;
     private JMSClientConnector jmsClientConnector;
-    private CarbonMessage message;
+    private Message message;
 
     public JMSPublisher(String destination, Map<String, String> staticJMSProperties,
-            JMSClientConnector jmsClientConnector, Object payload) throws UnsupportedEncodingException {
+                        JMSClientConnector jmsClientConnector, Object payload)
+            throws UnsupportedEncodingException {
         this.jmsProperties = new HashMap<>();
         this.jmsProperties.putAll(staticJMSProperties);
         this.jmsProperties.put(JMSConstants.PARAM_DESTINATION_NAME, destination);
         this.jmsClientConnector = jmsClientConnector;
-        this.message = handleCarbonMessage(payload);
+        try {
+            this.message = handleMessage(payload);
+        } catch (JMSConnectorException | JMSException e) {
+            log.error("Failed to process payload: " + e.getMessage());
+        }
     }
 
     @Override
     public void run() {
         try {
-            jmsClientConnector.send(message, null, jmsProperties);
-        } catch (ClientConnectorException e) {
-            log.error("Error sending JMS message: ", e);
+            jmsClientConnector.send(message, jmsProperties.get(JMSConstants.PARAM_DESTINATION_NAME));
+        } catch (JMSConnectorException e) {
+            log.error("Error sending JMS message: " + e.getMessage());
         }
     }
 
-    private CarbonMessage handleCarbonMessage(Object payload) throws UnsupportedEncodingException {
+    private Message handleMessage(Object payload) throws UnsupportedEncodingException,
+            JMSConnectorException, JMSException {
         if (payload instanceof String) {
-            CarbonMessage textCarbonMessage = new TextCarbonMessage(payload.toString());
-            this.jmsProperties.put(MESSAGE_TYPE_FIELD, TEXT_MESSAGE_TYPE);
-            return textCarbonMessage;
+            TextMessage message = (TextMessage) jmsClientConnector.createMessage(JMSConstants.TEXT_MESSAGE_TYPE);
+            message.setText(payload.toString());
+            return message;
         } else if (payload instanceof Map) {
-                MapCarbonMessage mapCarbonMessage = new MapCarbonMessage();
-                ((Map) payload).forEach((key, value) -> {
-                    mapCarbonMessage.setValue((String) key, String.valueOf(value));
-                });
-                return mapCarbonMessage;
+            MapMessage message = (MapMessage) jmsClientConnector.createMessage(JMSConstants.MAP_MESSAGE_TYPE);
+            ((Map) payload).forEach((key, value) -> {
+                try {
+                    message.setStringProperty((String) key, String.valueOf(value));
+                } catch (JMSException e) {
+                    log.error("Error while adding into message properties. " + e.getMessage());
```

Comment:
We should not just log and continue here because incase JMSExecption occurs, we will be sending an empty message out. We need to propogate this exception up.

Reply:
Fixed with commit [#3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc](https://github.com/wso2-extensions/siddhi-io-jms/pull/25/commits/3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc)